### PR TITLE
Alerting: Remove legacy alerting

### PR DESCRIPTION
**Pull request by gillesdemey**
**What is this feature?**

This PR removes legacy Alerting from Grafana; this PR is very large so some helpful tips to review (if you feel inclined).

1. Filter files by extensions that you are comfortable with to review (`*.ts(x)`, `*.go`, `*.md`, etc)
2. Filter by "files owned by me"

**Special notes for your reviewer:**

Some things the type checker won't catch and require human review;

1. Dynamic imports using `React.lazy()`
3. Changes made to the core redux selector could use special attention

### CODE REVIEW

File: .github/CODEOWNERS
Comment (gillesdemey) [inline .github/CODEOWNERS:1]: `/pkg/services/alerting/` is still in the trunk – slated to be removed from this PR

File: pkg/services/alerting/engine.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ type AlertEngine struct {
 
 // IsDisabled returns true if the alerting service is disabled for this instance.
 func (e *AlertEngine) IsDisabled() bool {
-	return e.Cfg.AlertingEnabled == nil || !*(e.Cfg.AlertingEnabled) || !e.Cfg.ExecuteAlerts || e.Cfg.UnifiedAlerting.IsEnabled()
+	return true
```
Comment (gillesdemey) [inline pkg/services/alerting/engine.go:60]: Do we need to follow-up in this PR to remove all calls to `AlertEngine.isDisabled()` or future work?

File: pkg/services/alerting/notifier.go [hunk 2]
```diff
@@ -219,7 +219,7 @@ func (n *notificationService) renderAndUploadImage(evalCtx *EvalContext, timeout
 		},
 		Width:           1000,
 		Height:          500,
-		ConcurrentLimit: n.cfg.AlertingRenderLimit,
+		ConcurrentLimit: 0,
```
Comment (gillesdemey) [inline pkg/services/alerting/notifier.go:222]: This is just to satisfy the compiler for now – when we remove `pkg/services/alerting` this should no longer be of any concern.

File: pkg/services/provisioning/provisioning.go [hunk 3]
```diff
@@ -231,12 +225,7 @@ func (ps *ProvisioningServiceImpl) ProvisionPlugins(ctx context.Context) error {
 }
 
 func (ps *ProvisioningServiceImpl) ProvisionNotifications(ctx context.Context) error {
-	alertNotificationsPath := filepath.Join(ps.Cfg.ProvisioningPath, "notifiers")
-	if err := ps.provisionNotifiers(ctx, ps.Cfg, alertNotificationsPath, ps.alertingService, ps.orgService, ps.EncryptionService, ps.NotificationService); err != nil {
-		err = fmt.Errorf("%v: %w", "Alert notification provisioning error", err)
-		ps.log.Error("Failed to provision alert notifications", "error", err)
-		return err
-	}
+	// TODO: Remove this entirely after dependencies are removed in enterprise.
```
Comment (gillesdemey) [inline pkg/services/provisioning/provisioning.go:228]: Follow-up in this PR or separate one?

**Review by marefr — APPROVED** [review]
Wow yes nice 🎉🙌  Files owned by us / @grafana/plugins-platform-backend, pkg/plugins and pkg/services/pluginsintegration LGTM

**Comment by marefr** [comment]
Looking forward to delete most of things in https://github.com/grafana/grafana/tree/main/pkg/tsdb/legacydata when this has been merged 💥

**Comment by torkelo** [comment]
I took this for a spin, looks good!

**Review by torkelo — APPROVED** [review]

**Pull request merged**